internal/build/lifecycle_execution.go

```diff
@@ -0,0 +1,421 @@
+package build
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/auth"
+	"github.com/docker/docker/client"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/internal/builder"
+	"github.com/buildpacks/pack/internal/cache"
+	"github.com/buildpacks/pack/internal/paths"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/logging"
+)
+
+type LifecycleExecution struct {
+	logger       logging.Logger
+	docker       client.CommonAPIClient
+	platformAPI  *api.Version
+	layersVolume string
+	appVolume    string
+	os           string
+	mountPaths   mountPaths
+	opts         LifecycleOptions
+}
+
+func NewLifecycleExecution(logger logging.Logger, docker client.CommonAPIClient, opts LifecycleOptions) (*LifecycleExecution, error) {
+	latestSupportedPlatformAPI, err := findLatestSupported(append(
+		opts.Builder.LifecycleDescriptor().APIs.Platform.Deprecated,
+		opts.Builder.LifecycleDescriptor().APIs.Platform.Supported...,
+	))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is the main change, where we determine the right platform API version to use.